frontend/src/routes/(authenticated)/project/[project_code]/BulkAddProjectMembers.svelte

```diff
@@ -0,0 +1,151 @@
+  import Icon from '$lib/icons/Icon.svelte';
+  import BadgeList from '$lib/components/Badges/BadgeList.svelte';
+  import { distinct } from '$lib/util/array';
+
+  enum BulkAddSteps {
+    Add,
+    Results,
+  }
+
+  let currentStep = BulkAddSteps.Add;
+
+  export let projectId: string;
+  const schema = z.object({
+    usernamesText: z.string().min(1, $t('register.name_missing')),
+    password: passwordFormRules($t),
+  });
+
+  let formModal: FormModal<typeof schema>;
+  $: form = formModal?.form();
+
+  let addedMembers: BulkAddProjectMembersResult['addedMembers'] = [{username: 'test2', 'role': ProjectRole.Manager}];
+  let createdMembers: BulkAddProjectMembersResult['createdMembers'] = [{username: 'test', 'role': ProjectRole.Editor}];
+  let existingMembers: BulkAddProjectMembersResult['existingMembers'] = [{username: 'test2', 'role': ProjectRole.Manager}];
+  $: addedCount = addedMembers.length + createdMembers.length;
+
+  const usernameRe = /^[a-zA-Z0-9_]+$/;
+
+  function validateUsernames(usernames: string[]): boolean {
+    return usernames.every(s => usernameRe.test(s));
+  }
+
+  async function openModal(): Promise<void> {
+    currentStep = BulkAddSteps.Add;
+    const { response } = await formModal.open(undefined, async (state) => {
+      const passwordHash = await hash(state.password.currentValue);
+      const usernames = state.usernamesText.currentValue
+        .split('\n')
+        // Remove whitespace
+        .map(s => s.trim())
+        // Remove empty lines before validating, otherwise final newline would count as invalid because empty string
+        .filter(s => s)
+        .filter(distinct);
+      if (!validateUsernames(usernames)) {
+        return $t('project_page.bulk_add_members.usernames_alphanum_only');
+      }
+      const { error, data } = await _bulkAddProjectMembers({
+        projectId,
+        passwordHash,
+        usernames,
+        role: ProjectRole.Editor, // Managers not allowed to have shared passwords
+      });
+
+      addedMembers = data?.bulkAddProjectMembers.bulkAddProjectMembersResult?.addedMembers ?? [];
+      createdMembers = data?.bulkAddProjectMembers.bulkAddProjectMembersResult?.createdMembers ?? [];
+      existingMembers = data?.bulkAddProjectMembers.bulkAddProjectMembersResult?.existingMembers ?? [];
+      return error?.message;
+    }, { keepOpenOnSubmit: true });
+
+    if (response === DialogResponse.Submit) {
+      currentStep = BulkAddSteps.Results;
+    }
+  }
+</script>
+
+<AdminContent>
+  <BadgeButton type="badge-success" icon="i-mdi-account-plus-outline" on:click={openModal}>
+    {$t('project_page.bulk_add_members.add_button')}
+  </BadgeButton>
+
+  <FormModal bind:this={formModal} {schema} let:errors>
+    <span slot="title">{$t('project_page.bulk_add_members.modal_title')}</span>
+    {#if currentStep == BulkAddSteps.Add}
+      <p>{$t('project_page.bulk_add_members.explanation')}</p>
+      <Input
+        id="password"
```

Comment:
firefox tries to auto fill my password here because of the id, can we just remove it so that doesn't happen?

Reply:
Done in commit 89648eda5ead95917c86a184bb47340961f74dfb.